src/lib/redux/inventoryData.ts

```diff
@@ -136,3 +138,29 @@ export const setCurrentInventoryIdInRedux = (inventoryId: string): void => {
 export const addProductToRedux = (product: ProductRecord): void => {
   store.dispatch(addProduct(product));
 };
+
+export const selectAmountPurchaseRequestedApproved = createSelector(
+    selectAllCurrentSitePurchaseRequestsArray,
+    (purchaseRequests) => {
+      let amount = 0;
+      purchaseRequests.forEach((pr: PurchaseRequestRecord) => {
+        if (pr.status === PurchaseRequestStatus.APPROVED) {
+          amount += pr.amountSpent;
+        }
+      });
+      return amount;
+    }
+)
+
+export const selectAmountPurchaseRequestedDenied = createSelector(
+  selectAllCurrentSitePurchaseRequestsArray,
```

Comment:
Make sure that you calculate the amount denied for just the current period not all periods